src/slider/slider.js

```diff
@@ -0,0 +1,225 @@
+  domNode: ?Element | Text;
+  onMouseUpListener: ?MouseEventHandler;
+  onMouseMoveListener: ?MouseEventHandler;
+  static defaultProps = {
+    overrides: {},
+    onChange: () => {},
+    onBlur: () => {},
+    onFocus: () => {},
+    onMouseEnter: () => {},
+    onMouseLeave: () => {},
+    onMouseDown: () => {},
+    onMouseUp: () => {},
+    error: false,
+    autoFocus: false,
+    tabIndex: 0,
+  };
+
+  state = {
+    isThumbMoving: false,
+    currentThumb: -1,
+    currentMove: 0,
+    thumbRefs: this.props.value.map(() => React.createRef()),
+  };
+
+  constructor(props: PropsT) {
+    super(props);
+  }
+
+  onFocus = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onBlur = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onMouseEnter = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onMouseLeave = (e: SyntheticEvent<HTMLInputElement>) => {};
+
+  onChange = (e: SyntheticEvent<HTMLInputElement>, type: ChangeActionT) => {};
+
+  componentDidMount() {
+    /* eslint-disable-next-line react/no-find-dom-node */
+    this.domNode = findDOMNode(this);
+    this.addDocumentMouseEvents();
+  }
+
+  componentWillUnmount() {
+    if (super.componentWillUnmount) super.componentWillUnmount();
+    this.removeDocumentEvents();
+  }
+
+  onMouseDown = (e: MouseEvent, thumbIndex: number) => {
+    this.setState({
+      isThumbMoving: true,
+      currentThumb: thumbIndex,
+    });
+  };
+
+  onMouseUp = (e: MouseEvent) => {
+    if (this.state.isThumbMoving) {
+      this.setState({
+        isThumbMoving: false,
+        currentThumb: -1,
+        currentMove: 0,
+      });
+    }
+  };
+
+  onMouseMove = (e: MouseEvent) => {
+    if (this.state.isThumbMoving) {
+      this.onMove(e.movementX, e);
+    }
+  };
+
+  addDocumentMouseEvents() {
+    if (__BROWSER__) {
+      this.onMouseMoveListener = document.addEventListener(
+        'mousemove',
+        this.onMouseMove,
+      );
+      this.onMouseUpListener = document.addEventListener(
+        'mouseup',
+        this.onMouseUp,
+      );
+    }
+  }
+
+  removeDocumentEvents() {
+    this.onMouseMoveListener && this.onMouseMoveListener.remove();
+    this.onMouseUpListener && this.onMouseUpListener.remove();
+  }
+
+  render() {
+    const {overrides = {}, range, value} = this.props;
+    const {thumbRefs} = this.state;
+    const [Root, rootProps] = getOverrides(overrides.Root, StyledRoot);
+    const [Axis, axisProps] = getOverrides(overrides.Axis, StyledAxis);
+    const [AxisRange, axisRangeProps] = getOverrides(
+      overrides.AxisRange,
+      StyledAxisRange,
+    );
+    const [Thumb, thumbProps] = getOverrides(overrides.Thumb, StyledThumb);
+    const [Tick, tickProps] = getOverrides(overrides.Tick, StyledTick);
+    const [TickBar, tickBarProps] = getOverrides(
+      overrides.TickBar,
+      StyledTickBar,
+    );
+    const sharedProps = this.getSharedProps();
+    return (
+      <Root {...sharedProps} onClick={e => this.onAxisClick(e)} {...rootProps}>
```

Comment:
Wonder if we should we move the `onClick` handler to go after `rootProps` but call a customer's `onClick` handler within it. What do you think?